src/view/edit-point-view.js

```diff
@@ -99,45 +101,119 @@ function createEditPointTemplate (point, allOffers, destinations ) {
           ${pointDestination.description ?
       `<h3 class="event__section-title  event__section-title--destination">Destination</h3>
            <p class="event__destination-description">${pointDestination.description}</p>
+
          </section>`
       : ''}
+      <div class="event__photos-container">
+      <div class="event__photos-tape">
+        ${pictures.map(({ src, description }) => `<img class="event__photo" src="${src}.jpg" alt="${description}">`).join('')}
+      </div>
+    </div>
 
         </section>
       </form>
     </li>`
   );
 }
-export default class EditPointView extends AbstractView {
+export default class EditPointView extends AbstractStatefulView {
 
   #point = null;
   #allOffers = null;
   #destinations = null;
   #handleFormSubmit = null;
   #handleEditClick = null;
 
+
   constructor({ point, allOffers, destinations, onFormSubmit, onEditClick}) {
     super();
     this.#point = point;
     this.#allOffers = allOffers;
     this.#destinations = destinations;
     this.#handleFormSubmit = onFormSubmit;
     this.#handleEditClick = onEditClick;
-    this.element.querySelector('form').addEventListener('submit', this.#formSubmitHandler);
-    this.element.querySelector('.event__rollup-btn').addEventListener('click', this.#editBtnHandler);
+    this._setState(EditPointView.parsePointToState(point));
+    this.#setInnerHandlers();
+
   }
 
   get template() {
-    return createEditPointTemplate(this.#point, this.#allOffers, this.#destinations);
+    return createEditPointTemplate(this._state, this.#allOffers, this.#destinations);
   }
 
+  static parsePointToState = (point) => ({ ...point });
+
+  static parseStateToPoint = (state) => ({ ...state });
+
   #formSubmitHandler = (evt) => {
     evt.preventDefault();
-    this.#handleFormSubmit();
+    this.#handleFormSubmit(EditPointView.parseStateToPoint(this._state));
   };
 
   #editBtnHandler = (evt) => {
     evt.preventDefault();
     this.#handleEditClick();
   };
 
+  #pointTypeChangeHandler = (evt) => {
+    evt.preventDefault();
+
+    if (evt.target.tagName === 'INPUT') {
+      this.updateElement({
+        type: evt.target.value,
+        offers: []
+      });
+    }
+  };
+
+
+  #destinationChangeHandler = (evt) => {
+    evt.preventDefault();
+
+    if (!evt.target.value) {
+      this.updateElement({
+        destination: ''
+      });
+      return;
+    }
+    const selectedDestination = this.#destinations
```

Comment:
Можно использовать деструктуризацию 
```suggestion
    if (!evt.target.value) {
      this.updateElement({
        destination: ''
      });
      return;
    }
    const { id } = this.#destinations
```